### PR TITLE
[DO NOT MERGE] Trying to use ajax in main.js

### PR DESCRIPTION
**Pull request by championswimmer**
This does not work as of now. 
Just sending a simple res.sendStatus work also does not work. 

Will have to figure out what the problem is.

### CODE REVIEW

File: src/generator/backend/generator.js [hunk 1]
```diff
@@ -162,26 +162,32 @@ exports.pipeZipToRes = function(req, res) {
...
-        throw err;
-      });
-
-      zipfile.pipe(res);
-
-      zipfile.directory(distHelper.distPath, '/').finalize();
-      done(null, 'zip');
+      fs.writeFileSync(distHelper.distPath + '/index.html', tpl(getJsonData()));
+      fs.writeFileSync(distHelper.distPath + '/tracks.html', trackstpl(getJsonData()));
+      res.sendStatus(200)
```
Comment (juslee) [inline src/generator/backend/generator.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166232/issues/source?bid=3233857&fileBranchId=3359574#l168)

File: src/generator/app.js [hunk 2]
```diff
@@ -23,10 +23,9 @@ app.set('port', (process.env.PORT || 5000));
 app.use('/', express.static(__dirname + '/www'));
 
 app.post('/generate', uploadedFiles, function(req, res, next) {
-  // console.log(req.files);
-  // console.log(req.body);
-  res.setHeader('Content-Type', 'application/zip');
-  generator.pipeZipToRes(req, res);
+  console.log(req.files);
+  console.log(req.body);
```
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166229/issues/source?bid=3233857&fileBranchId=3359574#l27)

File: src/generator/app.js [hunk 3]
```diff
@@ -23,10 +23,9 @@ app.set('port', (process.env.PORT || 5000));
 app.use('/', express.static(__dirname + '/www'));
 
 app.post('/generate', uploadedFiles, function(req, res, next) {
-  // console.log(req.files);
-  // console.log(req.body);
-  res.setHeader('Content-Type', 'application/zip');
-  generator.pipeZipToRes(req, res);
+  console.log(req.files);
```
Comment (juslee) [inline src/generator/app.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166229/issues/source?bid=3233857&fileBranchId=3359574#l26)

File: src/generator/www/js/main.js [hunk 4]
```diff
@@ -1,54 +1,32 @@
...
+    $.ajax({
+      url : '/generate',
+      type: 'POST',
+      method: 'POST',
+      data: form,
+      mimeType: 'multipart/form-data',
+      success: function(data, textStatus, jqXHR) {
+        console.log(data.status);
+      },
+      error: function (jqXHR, textStatus, errorThrown) {
```
Comment (juslee) [inline src/generator/www/js/main.js]: Codacy Issue found: ['errorThrown' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166233/issues/source?bid=3233857&fileBranchId=3359574#l26)

File: src/generator/www/js/main.js [hunk 5]
```diff
@@ -1,54 +1,32 @@
...
+    // form.append('trackfile', {'0': {}});
+    // form.append('locationfile', {'0': {}});
+
+    $.ajax({
+      url : '/generate',
+      type: 'POST',
+      method: 'POST',
+      data: form,
+      mimeType: 'multipart/form-data',
+      success: function(data, textStatus, jqXHR) {
```
Comment (juslee) [inline src/generator/www/js/main.js]: Codacy Issue found: ['jqXHR' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166233/issues/source?bid=3233857&fileBranchId=3359574#l23)

File: src/generator/www/js/main.js [hunk 6]
```diff
@@ -1,54 +1,32 @@
...
-    contentType: false,
-    
-    success :function(data){
-      console.log(obj);
-    }
-       
+$(document).ready( function() {
+
+  $('#btn').click( function() {
+    var form = new FormData($('form').get(0));
```
Comment (juslee) [inline src/generator/www/js/main.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166233/issues/source?bid=3233857&fileBranchId=3359574#l4)

File: src/generator/www/js/main.js (hunk 5 above)
Comment (juslee) [inline src/generator/www/js/main.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166233/issues/source?bid=3233857&fileBranchId=3359574#l23)

Comment (juslee) [inline src/generator/www/js/main.js]: Codacy Issue found: [Identifier 'jqXHR' does not match the pattern '^[a-z]+([A-Z][a-z]+)*$'.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166233/issues/source?bid=3233857&fileBranchId=3359574#l23)

File: src/generator/www/js/main.js (hunk 4 above)
Comment (juslee) [inline src/generator/www/js/main.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166233/issues/source?bid=3233857&fileBranchId=3359574#l26)

Comment (juslee) [inline src/generator/www/js/main.js]: Codacy Issue found: [Using "FunctionExpression" is not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2758166233/issues/source?bid=3233857&fileBranchId=3359574#l26)

**Comment by championswimmer** [comment]
will redo later

**Pull request closed without merging**